crates/db_queries/src/lib.rs

```diff
@@ -219,6 +221,22 @@ pub fn diesel_option_overwrite(opt: &Option<String>) -> Option<Option<String>> {
   }
 }
 
+pub fn diesel_option_overwrite_to_url(
+  opt: &Option<String>,
+) -> Result<Option<Option<lemmy_db_schema::Url>>, url::ParseError> {
+  Ok(match opt {
+    // An empty string is an erase
+    Some(unwrapped) => {
+      if !unwrapped.eq("") {
+        Some(Some(url::Url::parse(unwrapped)?.into()))
```

Comment:
Instead of doing the `?` here, do the `is_err` check from the previous one, (or a map_err), and do `return Err(ApiError::err("invalid_url").into());`

Reply:
Oh nm I see you got it, my bad.